data/api.ts

```diff
@@ -15,3 +15,7 @@ export async function postEndpoint(url: string, data: any) {
 export async function putEndpoint(url: string, data: any) {
   return fetchEndpoint(url, 'PUT', data)
 }
+
+export async function deleteEndpoint(url: string, data: any) {
+  return fetchEndpoint(url, 'DELETE', data)
```

Comment:
We shouldn't be including a `body` in a DELETE request. You'll need to modify `fetchEndpoint` to make `data` optional and only include the `body` in the request if `data` is provided